packages/elements/list/src/getListDeleteForward.ts

```diff
@@ -22,15 +30,52 @@ export const getListDeleteForward = (editor: SPEditor) => {
   const res = getListItemEntry(editor, {});
 
   let moved: boolean | undefined = false;
-  if (!isSelectionAtBlockEnd(editor) || !res) {
+  if (!isSelectionAtBlockEnd(editor)) {
     return moved;
   }
 
+  if (!res) {
+    const afterPoint = Editor.after(editor, editor.selection!.focus.path);
+
+    if (afterPoint) {
+      // there is a block after it
+      const nextSiblingListRes = getListItemEntry(editor, {
+        at: afterPoint.path,
+      });
+
+      if (nextSiblingListRes) {
+        // the next block is a list
+        const { listItem } = nextSiblingListRes;
+        const parentBlockEntity = getBlockAbove(editor, {
+          at: editor.selection!.anchor,
+        });
+
+        if (!getText(editor, parentBlockEntity![1])) {
+          // the selected block is empty
+          Transforms.removeNodes(editor, { at: editor.selection!.focus });
+
+          return true;
+        }
+
+        if (hasListChild(editor, listItem[0])) {
+          // the next block has children, so we have to move the first item up
+          const sublistRes = getListItemEntry(editor, {
+            at: [...listItem[1], 1, 0, 0],
+          });
+
+          moveListItemUp(editor, sublistRes!);
+        }
+      }
+    }
+    return false;
+  }
+
   Editor.withoutNormalizing(editor, () => {
     const { listItem } = res;
 
+    // if it has no children
     if (!hasListChild(editor, listItem[0])) {
-      const li = getPlatePluginOptions(editor, ELEMENT_LI);
+      const liType = getPlatePluginType(editor, ELEMENT_LI);
```

Comment:
Any specific reason to use `getPlatePluginType` instead of options?

Reply:
It just didn't seem necessary to get all options when the only thing we use is the type and there is a helper function for it and almost always that is the one that is used in the repo as far as I have seen so far.

Reply:
Fair enough. I kind of prefer working with objects when possible so as to be able to check instances, force of habit. With typescript being able to brand strings, it is less of a thing. Plate doesn't do branded types anyway, so whatever is the prevalent approach in the codebase is fine with me.